pkg/upgrade/machine_deployment.go

```diff
@@ -96,18 +95,9 @@ func (u *MachineDeploymentUpgrader) updateMachineDeployment(machineDeployment *c
 	}
 
 	// Get the updated version in json
-	updated, err := json.Marshal(machineDeployment)
-	if err != nil {
-		return errors.Wrap(err, "error marshaling updated machine deployment to json")
-	}
-
-	// Create the patch
-	patchBytes, err := jsonpatch.CreateMergePatch(original, updated)
-	if err != nil {
-		return errors.Wrap(err, "error creating json patch")
-	}
+	updated := machineDeployment.DeepCopy()
 
-	_, err = u.managementClusterAPIClient.MachineDeployments(u.clusterNamespace).Patch(machineDeployment.Name, types.MergePatchType, patchBytes)
+	err := u.ctrlClient.Patch(context.TODO(), updated, ctrlclient.MergeFrom(original))
```

Comment:
instead of comparing against a deep copy this could compare against machineDeployment directly.

General convention we are using is that if a call returns just one error then it should be in the form of

```
if err := fn(); err != nil {
...
}
```

The benefit here is that we avoid the problem of accidentally not checking errors since this creates a new block. We don't run into:

```
x, err := fn()
// oops forgot to check error here but the compiler doesn't care because `err` is not an unused variable
y, err =  fn2()
if err != nil {}
```